test/index.spec.js

```diff
@@ -4,24 +4,28 @@
 const chai = require('chai')
 const { expect } = chai
 chai.use(require('dirty-chai'))
-const IPFSFactory = require('ipfsd-ctl')
+const Ctl = require('ipfsd-ctl')
 const PeerID = require('peer-id')
 
 const DelegatedPeerRouting = require('../src')
-const factory = IPFSFactory.create({ type: 'go' })
+const factory = Ctl.createFactory({ type: 'go', test: true})
```

Comment:
```suggestion
const factory = Ctl.createFactory(
  {
    type: 'go',
    test: true,
    ipfsHttpModule: require('ipfs-http-client')
  },
  {
    go: {
      ipfsBin: require('go-ipfs-dep').path()
    },
    js: {
      ipfsBin: require.resolve('ipfs/src/cli/bin.js')
    }
  })
```

to run type: 'js' you should add ipfs to dev deps